lib/lotus/action/cookie_jar.rb

```diff
@@ -131,6 +137,7 @@ def extract(env)
         hash   = env[COOKIE_HASH_KEY] ||= {}
         string = env[HTTP_HEADER]
 
+        return if hash.has_key? RACK_SESSION_KEY
```

Comment:
In a previous version I checked string instead hash with has_key, I'll try to update test with string version and adding more cookies in tests